subsys/nrf_rpc/rpc_utils/server/dev_info.c

```diff
@@ -27,3 +222,67 @@ static void get_server_version(const struct nrf_rpc_group *group, struct nrf_rpc
 
 NRF_RPC_CBOR_CMD_DECODER(rpc_utils_group, get_server_version, RPC_UTIL_DEV_INFO_GET_VERSION,
 			 get_server_version, NULL);
+
+static void get_crash_info(const struct nrf_rpc_group *group, struct nrf_rpc_cbor_ctx *ctx,
+			   void *handler_data)
+{
+	int rc;
+	struct nrf_rpc_cbor_ctx rsp_ctx;
+	struct nrf_rpc_crash_info info = { 0 };
+
+	nrf_rpc_cbor_decoding_done(group, ctx);
+
+	rc = load_crash_info(&info);
+	if (rc) {
+		NRF_RPC_CBOR_ALLOC(group, rsp_ctx, 1);
+
+		nrf_rpc_encode_null(&rsp_ctx);
+	} else {
+		NRF_RPC_CBOR_ALLOC(group, rsp_ctx, DUMP_METADATA_SIZE(info));
+
+		nrf_rpc_encode_uint(&rsp_ctx, info.uuid);
+		nrf_rpc_encode_uint(&rsp_ctx, info.reason);
+		nrf_rpc_encode_uint(&rsp_ctx, info.pc);
+		nrf_rpc_encode_uint(&rsp_ctx, info.lr);
+		nrf_rpc_encode_uint(&rsp_ctx, info.sp);
+		nrf_rpc_encode_uint(&rsp_ctx, info.xpsr);
+		nrf_rpc_encode_uint(&rsp_ctx, info.assert_line);
+		nrf_rpc_encode_str(&rsp_ctx, info.assert_filename, -1);
+	}
+
+	nrf_rpc_cbor_rsp_no_err(group, &rsp_ctx);
+}
+
+NRF_RPC_CBOR_CMD_DECODER(rpc_utils_group, get_crash_info, RPC_UTIL_DEV_INFO_GET_CRASH_INFO,
+			 get_crash_info, NULL);
+
+#ifndef CONFIG_ASSERT_NO_FILE_INFO
+void assert_post_action(const char *file, unsigned int line)
+{
+	next_dump_info.assert_line = line;
+	next_dump_info.header.has_assert_info = true;
+
+	strncpy(next_dump_info.assert_filename, file, sizeof(next_dump_info.assert_filename));
```

Comment:
note that `strncpy` does not add null-terminator if the input is larger than the buffer. Is the code prepared for that? Btw, I wonder if it wouldn't make sense to store the last part of the filename in such a case.